docs/pages/machine-id/guides/gitlab.mdx

```diff
@@ -0,0 +1,166 @@
+
+In order to allow your GitLab CI to authenticate to your Teleport cluster,
+you'll first need to create a join token. These tokens define the rules that
+allow a bot to join your Teleport cluster.
+
+A GitLab join token contains allow rules that describe which pipelines should
+be able to use that token in order to join the Teleport cluster. A rule can
+contain multiple constraints, and any pipeline that matches all of the
+constraints within a single rule is granted access.
+
+The following constraints exist:
+- `sub`: a string uniquely identifying the CI run's source. It follows the
+  following format:
+  `project_path:{group}/{project}:ref_type:{type}:ref:{branch_name}`
+- `ref`: the git reference that triggered the CI run
+- `ref_type`: the type of reference that triggered the CI run. This will either
+  be `branch` or `tag`
+- `namespace_path`: the group or user that the project belongs to.
+- `project_path`: the full name of the project that the run belongs to. For
+example: `my-user/my-project`
+- `pipeline_source`: the trigger of the CI run. For example: `web`
+- `environment`: the environment that the job is associated with
+
+A full explanation of these constraints values can be found on the [GitLab
+Documentation.](https://docs.gitlab.com/ee/ci/secrets/id_token_authentication.html#token-payload)
+
+In this example, you will create a rule that grants access to any GitLab CI job
+within a specific GitLab project. Determine the fully qualified path of your
+GitLab project. This will include your username (or group) and the name of your
+project, e.g `my-user/my-project`.
+
+Create a file named `gitlab-token.yaml`. Ensure you substitute any values as
+suggested by the comments in this example.
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: gitlab-demo
+spec:
+  roles: [Bot]
+  join_method: gitlab
+  bot_name: gitlab-demo
+  gitlab:
+    # domain should be the domain of your GitLab instance. If you are using
+    # GitLab's cloud hosted offering, omit this field entirely.
+    domain: gitlab.example.com
+    allow:
+        # project_path should be the fully qualified path of your GitLab
+        # project that you determined earlier.
+      - project_path: my-user/my-project
+```
+
+Apply this to your Teleport cluster using `tctl`:
+
+```code
+$ tctl create -f gitlab-token.yaml
+```
+
+## Step 2/3. Create a Machine ID bot
+
+With the join token created, the next step is to create the Machine ID bot that
+the token will grant access to.
+
+In this example, the bot is given the default `access` role. In a production
+environment, the principle of least privilege should be applied and you should
+create a role that grants the bot access to the precise resources that will be
+needed in your CI/CD pipeline.
+
+Enter the following command from your workstation with `tsh` access, replacing
+the `ci-user` login with a Linux user on the host that you want your GitLab
+pipeline to be able to connect to:
```

Comment:
Same comment as this section in `docs/pages/machine-id/guides/github-actions.mdx`.